plugins/modules/s3_metrics_configuration.py

```diff
@@ -0,0 +1,246 @@
+    except is_boto3_error_code('NoSuchConfiguration'):
+        metrics_configuration = None
+    except ClientError as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to get bucket metrics configuration")
+
+    if metrics_configuration:
+        if not _compare_metrics_configuration(metrics_configuration, mc_id, filter_prefix, filter_tags):
+            try:
+                if not module.check_mode:
+                    client.put_bucket_metrics_configuration(
+                        Bucket=bucket_name,
+                        Id=mc_id,
+                        MetricsConfiguration=_create_metrics_configuration(mc_id, filter_prefix, filter_tags)
+                    )
+                changed = True
+            except (BotoCoreError, ClientError) as e:
+                module.fail_json_aws(e, msg="Failed to udpate bucket metrics configuration '%s'" % mc_id)
+    else:
+        try:
+            if not module.check_mode:
+                client.put_bucket_metrics_configuration(
+                    Bucket=bucket_name,
+                    Id=mc_id,
+                    MetricsConfiguration=_create_metrics_configuration(mc_id, filter_prefix, filter_tags)
+                )
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Failed to create bucket metrics configuration '%s'" % mc_id)
+
+    module.exit_json(changed=changed)
+
+
+def delete_metrics_configuration(client, module):
+    changed = False
+    bucket_name = module.params.get('bucket_name')
+    mc_id = module.params.get('id')
+
+    try:
+        client.get_bucket_metrics_configuration(Bucket=bucket_name, Id=mc_id)
+    except is_boto3_error_code('NoSuchConfiguration'):
+        module.exit_json(changed=changed)
+    except ClientError as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to get bucket metrics configuration")
+
+    try:
+        if not module.check_mode:
+            client.delete_bucket_metrics_configuration(Bucket=bucket_name, Id=mc_id)
+        changed = True
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to delete bucket metrics configuration '%s'" % mc_id)
+
+    module.exit_json(changed=changed)
+
+
+def main():
+    argument_spec = dict(
+        bucket_name=dict(type='str', required=True),
+        id=dict(type='str', required=True),
+        filter_prefix=dict(type='str', required=False),
+        filter_tags=dict(default={}, type='dict', required=False, aliases=['filter_tag']),
+        state=dict(default='present', type='str', choices=['present', 'absent']),
+    )
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    state = module.params.get('state')
+
+    try:
+        client = module.client('s3', retry_decorator=AWSRetry.exponential_backoff(retries=10, delay=3))
```

Comment:
Using the retry_decorator is a really good thing to do, it's especially helpful with a busy account that might start hitting the rate limits (our CI does during 'full' runs for example).  As a quirk of how AWSRetry was originally written, even with the decorator configured you still need to explicitly request retries (It used to retry on certain 'NotFound' type errors, which wasn't the right thing to do for many calls)

Enabling the retries for any specific call is however *really* easy:
```
client.delete_bucket_metrics_configuration(aws_retry=True, ...)
```